### DISCUSSION TITLE
Is it possible to unmount a sub application ?

### DISCUSSION THREAD

**Discussion by TheophileDiot** [fastapi/fastapi#9995] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI

# my main app
app = FastAPI()

@app.get("/app")
def read_main():
    return {"message": "Hello World from main app"}

# my first sub app
subapi = FastAPI()

@subapi.get("/sub")
def read_sub():
    return {"message": "Hello World from sub API"}

app.mount("/subapi", subapi)

# my second sub app
subapi_v2 = FastAPI()

@subapi_v2.get("/sub")
def read_sub():
    return {"message": "Hello World from sub API V2"}

app.mount("/subapi", subapi_v2)
```

### Description

I'm wondering if there is a way to override/unmount mounted sub applications ?

I saw that you can dynamically add routes and remove them but mount are way easier to handle, especially when you don't know the routes in advance. Is there a way to achieve that ?

My current solution is to loop through `app.routes` and remove all the `Mount` objects from the list, but I'm not sure if this is the optimal way memory wise.

You can try my example by just going to `/subapi/sub` and see that the `Hello World from sub API` message is showing instead of the `Hello World from sub API V2` message.

### Operating System

Linux

### FastAPI Version

0.100.1

### Python Version

3.11.0rc1

**Comment by mattmess1221** [comment, marked as answer]
A mount is just a fancy route, so you can remove it from the routes.

```py
from starlette.routing import Mount

for index, route in enumerate(app.routes):
  if isinstance(route, Mount) and route.path == "/subapi":
    del app.routes[index]
    break
```
